crates/core/src/index/indexer.rs

```diff
@@ -71,38 +61,46 @@ impl<BE: DecryptWriteBackend> Indexer<BE> {
     /// # Arguments
     ///
     /// * `be` - The backend to write to.
-    pub fn new_unindexed(be: BE) -> Self {
+    pub fn new_unindexed() -> Self {
         Self {
-            be,
             file: IndexFile::default(),
             count: 0,
             created: SystemTime::now(),
             indexed: None,
         }
     }
 
-    /// Resets the indexer.
-    pub fn reset(&mut self) {
-        self.file = IndexFile::default();
-        self.count = 0;
-        self.created = SystemTime::now();
-    }
-
     /// Returns a `SharedIndexer` to use in multiple threads.
     ///
     /// # Type Parameters
     ///
     /// * `BE` - The backend type.
-    pub fn into_shared(self) -> SharedIndexer<BE> {
-        Arc::new(RwLock::new(self))
+    pub fn into_shared(self) -> SharedIndexer {
+        SharedIndexer(Arc::new(RwLock::new(self)))
     }
 
     /// Finalizes the `Indexer`.
     ///
     /// # Errors
     ///
     /// * If the index file could not be serialized.
-    pub fn finalize(&self) -> RusticResult<()> {
+    pub fn finalize(mut self) -> Option<IndexFile> {
+        self.save()
+    }
+
+    pub fn needs_save(&self) -> bool {
+        // check if IndexFile needs to be saved
+        let elapsed = self.created.elapsed().unwrap_or_else(|err| {
+            warn!("couldn't get elapsed time from system time: {err:?}");
```

Comment:
This should be part of the error handling here and not a warning in the logs.

Reply:
This is the error handling. When we can't get an elapsed time when we check if an index file should be saved, we have two options:
1) return an error which would mean we don't finish the backup
2) omit the check on the elapsed time, i.e. only write when the index file is "full" and warn the user.  This will successfully finish the backup.

I strongly prefer option 2) which is the current error handling.

Reply:
This wasn't what I meant, what I meant was if we have a warning in the log, user of rustic_core are not able to handle this, instead of a log warning this should return a rustic_error with the message so a user can handle this if we get to it from somewhere.